staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -193,6 +196,31 @@ func (c *namedCreaterAdapter) Create(ctx context.Context, name string, obj runti
 	return c.Creater.Create(ctx, obj, createValidatingAdmission, options)
 }
 
+// manager is assumed to be already a valid value, we need to make
+// userAgent into a valid value too.
+func managerOrUserAgent(manager, userAgent string) string {
+	if manager != "" {
+		return manager
+	}
+	return prefixFromUserAgent(userAgent)
+}
+
+// prefixFromUserAgent takes the characters preceding the first /, quote
+// unprintable character and then trim what's beyond the
+// FieldManagerMaxLength limit.
 func prefixFromUserAgent(u string) string {
-	return strings.Split(u, "/")[0]
+	m := strings.Split(u, "/")[0]
+	buf := bytes.NewBuffer(nil)
+	for _, r := range m {
```

Comment:
You may want to go here: https://blog.golang.org/strings and search for "range loops" to learn more surprising things about go. (just a note, not asking for a change)